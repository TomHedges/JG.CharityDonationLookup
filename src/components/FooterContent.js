import React, { Component } from "react";
import { hot } from "react-hot-loader";
import PropTypes from "prop-types";
import "../styles/jgdr_styles.css";

class FooterContent extends Component {
  render() {
    return (
      <div className="footer_content side_pad">
        <p>Â© Tom Hedges 2019 - {this.props.heading}</p>
      </div>
    );
  }
}

FooterContent.propTypes = {
  heading: PropTypes.string.isRequired
};

export default hot(module)(FooterContent);
