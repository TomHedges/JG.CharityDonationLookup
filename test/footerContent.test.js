/**
 * Unit tests for the FooterContent component
 */

import FooterContent from "../src/components/FooterContent";

const HEADING = "JG Donations Report";

function createTestProps(props) {
  return {
    heading: HEADING,
    ...props
  };
}

let wrapper, props;

beforeEach(() => {
  props = createTestProps();
  wrapper = shallow(<FooterContent {...props} />);
});

describe("<FooterContent /> rendering", () => {
  it("Should render <FooterContent /> component", () => {
    expect(wrapper.exists()).toBe(true);
  });
});

describe("FooterContent should contain copyright assertion and header text", () => {
  it("Should display one <p>", () => {
    expect(wrapper.find("p")).toHaveLength(1);
  });

  it("Should display one <p> containing copyright assertion and heading", () => {
    expect(wrapper.find("p").text()).toEqual("Â© Tom Hedges 2019 - " + HEADING);
  });

  it("Snapshot should display copyright assertion and header text", () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
