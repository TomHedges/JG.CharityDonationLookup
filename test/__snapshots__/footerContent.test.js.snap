// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterContent should contain copyright assertion and header text Snapshot should display copyright assertion and header text 1`] = `
<div
  className="footer_content side_pad"
>
  <p>
    Â© Tom Hedges 2019 - 
    JG Donations Report
  </p>
</div>
`;
